src/Repositories/Body/MultipartBodyRepository.php

```diff
@@ -4,10 +4,159 @@
 
 namespace Saloon\Repositories\Body;
 
+use Saloon\Helpers\Arr;
+use InvalidArgumentException;
+use Saloon\Contracts\Arrayable;
+use Saloon\Data\MultipartValue;
+use Saloon\Traits\Conditionable;
+use Saloon\Contracts\Body\BodyRepository;
 use Saloon\Exceptions\UnableToCastToStringException;
 
-class MultipartBodyRepository extends ArrayBodyRepository
+class MultipartBodyRepository implements BodyRepository, Arrayable
 {
+    use Conditionable;
+
+    /**
+     * Base Repository
+     *
+     * @var \Saloon\Repositories\Body\ArrayBodyRepository
+     */
+    protected ArrayBodyRepository $data;
+
+    /**
+     * Constructor
+     *
+     * @param array<MultipartValue> $value
+     */
+    public function __construct(mixed $value = [])
+    {
+        $this->data = new ArrayBodyRepository;
+
+        $this->set($value);
+    }
+
+    /**
+     * Set a value inside the repository
+     *
+     * @param array<MultipartValue> $value
+     * @return $this
+     */
+    public function set(mixed $value): static
+    {
+        if (! is_array($value)) {
+            throw new InvalidArgumentException('The value must be an array');
+        }
+
+        $this->data->set(
+            $this->parseMultipartArray($value)
+        );
+
+        return $this;
+    }
+
+    /**
+     * Merge another array into the repository
+     *
+     * @param array<MultipartValue> ...$arrays
+     * @return $this
+     */
+    public function merge(array ...$arrays): static
+    {
+        $value = [];
+
+        foreach ($arrays as $array) {
+            $value[] = $this->parseMultipartArray($array);
+        }
+
+        $this->data->merge(...$value);
+
+        return $this;
+    }
```

Comment:
It should be possible to dumb this down, to a more declarative approach.

```suggestion
    public function merge(array ...$arrays): static
    {
        $this->data->merge(...array_map(
            $this->parseMultipartArray(...),
            $arrays,
        ));

        return $this;
    }
```